RecoHGCal/TICL/plugins/TICLSeedingRegionProducer.cc

```diff
@@ -0,0 +1,72 @@
+// Authors: Felice Pantaleo, Marco Rovere
+// Emails: felice.pantaleo@cern.ch, marco.rovere@cern.ch
+// Date: 06/2019
+
+#include <vector>
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "RecoHGCal/TICL/interface/SeedingRegionAlgoBase.h"
+#include "SeedingRegionByTracks.h"
+#include "SeedingRegionGlobal.h"
+
+using namespace ticl;
+
+class TICLSeedingRegionProducer : public edm::stream::EDProducer<> {
+public:
+  TICLSeedingRegionProducer(const edm::ParameterSet&);
+  ~TICLSeedingRegionProducer() override {}
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+private:
+  std::unique_ptr<SeedingRegionAlgoBase> myAlgo_;
+  int algoId_;
+  std::string seedingId_;
+};
+DEFINE_FWK_MODULE(TICLSeedingRegionProducer);
+
+TICLSeedingRegionProducer::TICLSeedingRegionProducer(const edm::ParameterSet& ps)
+    : algoId_(ps.getParameter<int>("algoId")) {
+  auto sumes = consumesCollector();
+
+  switch (algoId_) {
```

Comment:
eventually a factory should be used to construct algorithms by name on demand (can be addressed in a subsequent PR)